### PR TITLE
Add pom generation to heron-api

### PR DESCRIPTION
**Pull request by aahmed-se**
Make version tag externally configurable in future PR

### CODE REVIEW

File: heron/api/src/java/BUILD [hunk 1]
```diff
@@ -45,3 +45,16 @@ java_library(
     srcs = glob(["com/twitter/heron/classification/**/*.java"]),
     resources = ["//heron/api/src:processor-service-conf"],
 )
+
```
Comment (kramasamy) [inline heron/api/src/java/BUILD]: generate_pom.cc should go into tools - this looks a generation tool that can be shared by others.

Comment (aahmed-se) [inline heron/api/src/java/BUILD]: Will have to parameterize it to make it shareable , will do that.

File: heron/api/src/java/generate_pom.cc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#include <string>
+
+#define HERON_VERSION "0.14.6"
+
+int main(int argc, char** argv) {
+  std::string tmpl(R"EOF(<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.twitter.heron</groupId>
+  <artifactId>libtensorflow</artifactId>
```
Comment (objmagic) [inline heron/api/src/java/generate_pom.cc]: libtensorflow?

Comment (aahmed-se) [inline heron/api/src/java/generate_pom.cc]: Will change that and take a look at the existing pom scripts.

Comment (objmagic) [inline heron/api/src/java/generate_pom.cc]: Yes, we **must** follow existing POM files. The generated POM files **must** be the same one generated now.

Please read https://github.com/twitter/heron/pull/1811 if you want to know why.

And why a C++ program to do this job? A Python program should do well.

**Pull request closed without merging**